pkg/app/piped/driftdetector/lambda/detector.go

```diff
@@ -13,3 +13,385 @@
+	var (
+		apps = d.appLister.ListByPlatformProvider(d.provider.Name)
+		m    = make(map[string][]*model.Application)
+	)
+	for _, app := range apps {
+		repoID := app.GitPath.Repo.Id
+		m[repoID] = append(m[repoID], app)
+	}
+	return m
+}
+
+func (d *detector) checkApplication(ctx context.Context, app *model.Application, repo git.Repo, headCommit git.Commit) error {
+	headManifest, err := d.loadHeadFunctionManifest(app, repo, headCommit)
+	if err != nil {
+		return err
+	}
+	d.logger.Info(fmt.Sprintf("application %s has a function manifest at commit %s", app.Id, headCommit.Hash))
+
+	liveManifest, ok := d.stateGetter.GetFunctionManifest(app.Id)
+	if !ok {
+		return fmt.Errorf("failed to get a live function manifest of application %s", app.Id)
+	}
+	d.logger.Info(fmt.Sprintf("application %s has a live function manifest", app.Id))
+
+	ignoreAndSortParameters(&headManifest.Spec)
+
+	// WithIgnoreAddingMapKeys option ignores all of followings:
+	//  - default value of Architecture
+	//  - default value of EphemeralStorage
+	//  - environments added in live states
+	//  - tags added in live states, including pipecd managed tags
+	result, err := provider.Diff(
+		liveManifest,
+		headManifest,
+		diff.WithEquateEmpty(),
+		diff.WithIgnoreAddingMapKeys(),
+		diff.WithCompareNumberAndNumericString(),
+	)
+	if err != nil {
+		return err
+	}
+
+	state := makeSyncState(result, headCommit.Hash)
+
+	return d.reporter.ReportApplicationSyncState(ctx, app.Id, state)
+}
+
+// ignoreAndSortParameters removes parameters which cannot be compared and sorts specific parameters.
+// ignores:
+//   - SourceCode in headSpec
+//   - S3Bucket, S3Key, and S3ObjectVersion in headSpec
+//
+// sorts: (Lambda sorts them in liveSpec)
+//   - Architectures in headSpec
+//   - Environments in headSpec
+//   - SubnetIDs in headSpec
+//   - Tags in headSpec
+func ignoreAndSortParameters(headSpec *provider.FunctionManifestSpec) {
+	// We cannot compare SourceCode and S3 packaging because live states do not have them.
+	headSpec.SourceCode = provider.SourceCode{}
+	headSpec.S3Bucket = ""
+	headSpec.S3Key = ""
+	headSpec.S3ObjectVersion = ""
+
+	// Architectures, Environments, SubnetIDs, and Tags are sorted in live states.
+	if len(headSpec.Architectures) > 1 {
+		sort.Slice(headSpec.Architectures, func(i, j int) bool {
+			return strings.Compare(headSpec.Architectures[i].Name, headSpec.Architectures[j].Name) < 0
+		})
+	}
+	headSpec.Environments = sortMap(headSpec.Environments)
+	if headSpec.VPCConfig != nil && len(headSpec.VPCConfig.SubnetIDs) > 1 {
+		slices.Sort(headSpec.VPCConfig.SubnetIDs)
+	}
+	headSpec.Tags = sortMap(headSpec.Tags)
+}
+
+// sortMap sorts the given map by keys and returns a new map.
+func sortMap(m map[string]string) map[string]string {
+	keys := make([]string, 0, len(m))
+	for k := range m {
+		keys = append(keys, k)
+	}
+	slices.Sort(keys)
+
+	sorted := make(map[string]string)
+	for _, k := range keys {
+		sorted[k] = m[k]
+	}
+	return sorted
+}
```

Comment:
The for-range for map[string]string is always looped in random order.
So, this code may not work as expected.

Reply:
I understood, what a meaningless code... 😅 
let me re-think...

Reply:
@Warashi 
Thank you, I removed the unnecessary sort.  [be00911](https://github.com/pipe-cd/pipecd/pull/5186/commits/be009116934f76d32d349bc5258eb5f48d701440)

Some slices need to be sorted, but maps do NOT need to be sorted.